### PR TITLE
OpenEXR encoder: add capability to set the DWA compression level

### PR DESCRIPTION
**Pull request by reunanen**
extracted from #21324

(cherry picked from commit aa276ece66265dcf07af5f840933377d2106c871)

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/imgcodecs/src/grfmt_exr.cpp [hunk 1]
```diff
@@ -691,6 +691,10 @@ bool  ExrEncoder::write( const Mat& img, const std::vector<int>& params )
                 CV_Error(Error::StsBadArg, "IMWRITE_EXR_COMPRESSION is invalid or not supported");
             }
         }
+        if (params[i] == IMWRITE_EXR_DWA_COMPRESSION_LEVEL)
+        {
+            header.dwaCompressionLevel() = params[i + 1];
```
Comment (alalek) [inline modules/imgcodecs/src/grfmt_exr.cpp:696]: Need to add compile-time version check here (through `#if`)

Comment (reunanen) [reply modules/imgcodecs/src/grfmt_exr.cpp:698]: Oops, looks like the setting itself is available only from version 3 on.

Comment (reunanen) [reply modules/imgcodecs/src/grfmt_exr.cpp:697]: Should be ok now.

File: modules/imgcodecs/src/grfmt_exr.cpp [hunk 2]
```diff
@@ -691,6 +691,16 @@ bool  ExrEncoder::write( const Mat& img, const std::vector<int>& params )
                 CV_Error(Error::StsBadArg, "IMWRITE_EXR_COMPRESSION is invalid or not supported");
             }
         }
+        if (params[i] == IMWRITE_EXR_DWA_COMPRESSION_LEVEL)
+        {
+#if defined(OPENEXR_VERSION_MAJOR) && OPENEXR_VERSION_MAJOR >= 3
+            header.dwaCompressionLevel() = params[i + 1];
+#elif defined(OPENEXR_VERSION_MAJOR)
+            throw std::runtime_error("Setting `IMWRITE_EXR_DWA_COMPRESSION_LEVEL` not supported in OpenEXR version " + std::to_string(OPENEXR_VERSION_MAJOR) + " (version 3 is required)");
```
Comment (alalek) [inline modules/imgcodecs/src/grfmt_exr.cpp:699]: > `throw std::runtime_error`

Lets use `CV_LOG_ONCE_WARNING(NULL, "message");` instead to keep existed behavior for unknown/unsupported hints.

Comment (reunanen) [reply modules/imgcodecs/src/grfmt_exr.cpp:699]: Done

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Pull request merged**